### PR TITLE
refactor: use logbook, remove events code

### PR DESCRIPTION
**Pull request by b5**
Continuing work on logbook, this PR drops _events_ in favour of using the logbook.

### Rename to rev `dsref`, start splitting `repo.DatasetRef` into `dsref.Ref` and `repo.DatasetInfo`
* logbook should really live inside of the repo.Repo interface, which means logbook can't import the repo package to get at repo.DatasetRef. So, it's high time to make a change here, I've repurposed the `rev` package to be a place for working with dataset reference parsing
* I've added a shim method in `repo` for turning `repo.DatasetRef` into a `dsref.Ref` called `ConvertToDsref` to help with the transition. If we use this everywhere it'll be easy to spot the places where we'll need to update later
* the "meta" info in DatasetRef can start to transition to `repo.DatasetInfo`, which uses the `os.FileInfo` metaphor

### qfs changes
* I wanted to start logbook on newer apis only, so it makes use of an expanded `qfs.Filesystem` interface for storage, which now supports writes
* To make that work properly I've unified all the APIs that support writing to places, this means dropping the `pin` boolean argument on our old `cafs.Save`, which wasn't uniformly supported anyway. It's now assumed that if you `Put` something, you'd like to pin it. if you `Delete` something, you'd like to unpin it
* I've moved qfs filesystem muxing into the `qfs` package. It's super small & drops an import

### repo changes
* every repo now has either zero or one logbook
* it's ok to _not_ have a logbook. logbook's pointer methods must all handle being a pointer to `nil`, which means it's safe to call methods on a nil logbook. It'll return `logbook.ErrNoLogbook`
*

### CODE REVIEW

**Comment by b5** [comment]
@ramfox gonna need your review in particular, this breaks the `/log/` api, and will definitely affect Desktop. I'm happy to push this change back until later, but wanted to see if we can get away with the break in this PR first

**Comment by ramfox** [comment]
before looking at the code, I pulled, installed, and started playing with it. The first time I use `qri log REF` I get a list of references. The second time I use `qri log SAME_REF`, I get an empty list:

```
$ qri log ramfox/earthquakes_august_2019
1   2019-10-04T15:18:18Z        earthquakes_august_2019 /ipfs/Qmb3GMXhioY4jjb1aJcUjqGmrGxLZxE4cu3XmztYnZf91F    added to schema
2   2019-10-04T15:15:21Z        earthquakes_august_2019 /ipfs/QmciuaLQf8pNy2Ctzp75kEojNPYNmuxUX4TCQ99yr5Mw1Q    added a theme
3   2019-09-27T15:27:44Z        earthquakes_august_2019 /ipfs/QmSuuwq5sEMMotL481r24WS7ZJPv7fEuwLDwR7V4upowYa    Added keywords
4   2019-09-27T15:23:54Z        earthquakes_august_2019 /ipfs/QmZuVF7u86f36fZ1a1DFBZBC99KCRdHNqZ8a2CuS7SPC9H    Added keywords
5   2019-09-09T18:12:41Z        earthquakes_august_2019 /ipfs/QmYpg471bwzHw3oEeo6DCijrrDWwuRbdvaxmDtb2bTyD2d    Initial Commit

ramfox @ Kaseys-MacBook-Pro | ~/go/src/github.com/qri-io/qri (refactor_use_logbook)
$ qri log ramfox/earthquakes_august_2019
END
```

**Comment by ramfox** [comment]
In terms of API changes, the changes to the `/history/` endpoint response is v doable.

Logbook is clearing the way for indicating to the user that they have or do not have a certain version locally. How will this be indicated in the api? Another field on `DatasetLogItem`, such as `Available boolean`? Or some other way. Wondering if there is another big-ish refactor coming or if any changes to `DatasetLogItem` will be smaller. This will affect how we decide to implement the History response changes on desktop.

File: cmd/log_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	cmdR = r.CreateCommandRunner(ctx)
+	if err = executeCommand(cmdR, "qri logbook me/test_movies --raw"); err == nil {
+		t.Error("expected using a ref and the raw flag to error")
+	}
+
+	cmdR = r.CreateCommandRunner(ctx)
+	if err = executeCommand(cmdR, "qri logbook --raw"); err != nil {
+		t.Fatal(err)
+	}
+
```
[the remark this answers is not in the document]
Comment (b5) [reply cmd/log_test.go]: totally, not being able to override timestamps makes comparison tough 😢. I'll add a TODO & follow up later

File: lib/log_test.go [hunk 2]
```diff
@@ -58,11 +71,53 @@ func TestHistoryRequestsLog(t *testing.T) {
...
+
+	res := []LogEntry{}
+	if err = req.Logbook(&RefListParams{Ref: firstRef, Limit: 30}, &res); err != nil {
+		t.Fatal(err)
+	}
+
+	result := make([]string, len(res))
+	for i := range res {
+		// set response times to zero for consistent results
+		res[i].Timestamp = time.Time{}
```
[the remark this answers is not in the document]
Comment (b5) [reply lib/log_test.go]: totally. I'll add this to a subsequent PR

File: logbook/logbook_test.go [hunk 3]
```diff
@@ -137,28 +148,639 @@ func Example() {
 		panic(err)
 	}
 
-	for _, ref := range log {
-		fmt.Println(ref.String())
+	for _, info := range log {
+		fmt.Println(info.Ref.String())
```
[the remark this answers is not in the document]
Comment (b5) [reply logbook/logbook_test.go]: this is an "Example", the fmt.Println is captured by the test env & used in the `Output` comment as a check

**Pull request merged**